common/quotas/global/collection/internal/limiter.go

```diff
@@ -77,24 +72,22 @@ type (
 		//
 		// changing them to pointers will not change any semantics, so that should be fine if it becomes needed.
 
-		// accepted-request usage counter
-		accepted atomic.Int64
-		// rejected-request usage counter
-		rejected atomic.Int64
 		// number of failed updates, for deciding when to use fallback
 		failedUpdates atomic.Int64
+		// counts of what was allowed vs not
+		usage AtomicUsage
 
 		// ratelimiters in use
 
-		// fallback used when failedUpdates exceeds maxFailedUpdates,
-		// or prior to initial data from the global ratelimiter system.
-		fallback Limiter
-		// local-only limiter based global ratelimiter values.
+		// local-only limiter based global weight values.
 		//
 		// note that use and modification is NOT synchronized externally,
 		// so updates and deciding when to use the fallback must be done carefully
 		// to avoid undesired combinations when they interleave.
-		limit *rate.Limiter
+		primary clock.Ratelimiter
+		// fallback used when failedUpdates exceeds maxFailedUpdates,
+		// or prior to the first Update call with globally-adjusted values.
+		fallback quotas.Limiter
```

Comment:
separate interfaces because primary is settable, but fallback is not (it comes from a `*quotas.Collection` which returns only the `quotas.Limiter` API even though it's probably also a `clock.Ratelimiter` under the hood)